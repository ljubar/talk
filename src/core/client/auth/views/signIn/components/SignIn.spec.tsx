import { shallow } from "enzyme";
import { noop } from "lodash";
import React from "react";

import { removeFragmentRefs } from "talk-framework/testHelpers";
import { PropTypesOf } from "talk-framework/types";

import SignIn from "./SignIn";

const SignInN = removeFragmentRefs(SignIn);

it("renders correctly", () => {
  const props: PropTypesOf<typeof SignInN> = {
    onGotoSignUp: noop,
    emailEnabled: true,
    facebookEnabled: true,
    googleEnabled: true,
    oidcEnabled: true,
    auth: {},
  };
  const wrapper = shallow(<SignInN {...props} />);
  expect(wrapper).toMatchSnapshot();
});

it("renders without email login", () => {
  const props: PropTypesOf<typeof SignInN> = {
    onGotoSignUp: noop,
    emailEnabled: false,
    facebookEnabled: true,
    googleEnabled: true,
    oidcEnabled: true,
    auth: {},
  };
  const wrapper = shallow(<SignInN {...props} />);
  expect(wrapper).toMatchSnapshot();
});
