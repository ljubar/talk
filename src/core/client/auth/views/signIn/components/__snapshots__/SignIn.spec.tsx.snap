// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <Localized
    id="signIn-signInToJoinHeader"
    subtitle={<Subtitle />}
    title={<Title />}
  >
    <Bar>
      &lt;title&gt;Sign In&lt;/title&gt;&lt;subtitle&gt;to join the conversation&lt;/subtitle&gt;
    </Bar>
  </Localized>
  <SubBar>
    <Localized
      button={
        <withPropsOnChange(Button)
          color="primary"
          id="signIn-gotoSignUpButton"
          onClick={[Function]}
          size="small"
          variant="underlined"
        />
      }
      id="signIn-noAccountSignUp"
    >
      <withPropsOnChange(Typography)
        container={[Function]}
        variant="bodyCopy"
      >
        Don't have an account? &lt;button&gt;Sign Up&lt;/button&gt;
      </withPropsOnChange(Typography)>
    </Localized>
  </SubBar>
  <Main>
    <withPropsOnChange(HorizontalGutter)
      size="oneAndAHalf"
    >
      <withContext(createMutationContainer(withContext(createMutationContainer(SignInContainer)))) />
      <OrSeparator />
      <withPropsOnChange(HorizontalGutter)>
        <SignInWithFacebookContainer />
        <SignInWithGoogleContainer />
        <Relay(SignInWithOIDCContainer)
          auth={Object {}}
        />
      </withPropsOnChange(HorizontalGutter)>
    </withPropsOnChange(HorizontalGutter)>
  </Main>
</div>
`;

exports[`renders without email login 1`] = `
<div>
  <Localized
    id="signIn-signInToJoinHeader"
    subtitle={<Subtitle />}
    title={<Title />}
  >
    <Bar>
      &lt;title&gt;Sign In&lt;/title&gt;&lt;subtitle&gt;to join the conversation&lt;/subtitle&gt;
    </Bar>
  </Localized>
  <SubBar>
    <Localized
      button={
        <withPropsOnChange(Button)
          color="primary"
          id="signIn-gotoSignUpButton"
          onClick={[Function]}
          size="small"
          variant="underlined"
        />
      }
      id="signIn-noAccountSignUp"
    >
      <withPropsOnChange(Typography)
        container={[Function]}
        variant="bodyCopy"
      >
        Don't have an account? &lt;button&gt;Sign Up&lt;/button&gt;
      </withPropsOnChange(Typography)>
    </Localized>
  </SubBar>
  <Main>
    <withPropsOnChange(HorizontalGutter)
      size="oneAndAHalf"
    >
      <withPropsOnChange(HorizontalGutter)>
        <SignInWithFacebookContainer />
        <SignInWithGoogleContainer />
        <Relay(SignInWithOIDCContainer)
          auth={Object {}}
        />
      </withPropsOnChange(HorizontalGutter)>
    </withPropsOnChange(HorizontalGutter)>
  </Main>
</div>
`;
